src/coprocessor/dag/expr/ctx.rs

```diff
@@ -47,6 +47,136 @@ pub const FLAG_DIVIDED_BY_ZERO_AS_WARNING: u64 = 1 << 8;
 pub const MODE_ERROR_FOR_DIVISION_BY_ZERO: u64 = 27;
 
 const DEFAULT_MAX_WARNING_CNT: usize = 64;
+
+pub struct EvalConfigBuilder {
+    /// timezone
+    tz: FixedOffset,
+    ignore_truncate: bool,
+    truncate_as_warning: bool,
+    overflow_as_warning: bool,
+    in_insert_stmt: bool,
+    in_update_or_delete_stmt: bool,
+    in_select_stmt: bool,
+    pad_char_to_full_length: bool,
+    divided_by_zero_as_warning: bool,
+    max_warning_cnt: usize,
+    sql_mode: u64,
+    strict_sql_mode: bool,
+}
+
+impl EvalConfigBuilder {
+    pub fn new() -> EvalConfigBuilder {
+        EvalConfigBuilder {
+            tz: FixedOffset::east(0),
+            ignore_truncate: false,
+            truncate_as_warning: false,
+            overflow_as_warning: false,
+            in_insert_stmt: false,
+            in_update_or_delete_stmt: false,
+            in_select_stmt: false,
+            pad_char_to_full_length: false,
+            divided_by_zero_as_warning: false,
+            max_warning_cnt: DEFAULT_MAX_WARNING_CNT,
+            sql_mode: 0,
+            strict_sql_mode: false,
+        }
+    }
+
+    pub fn set_timezone(&mut self, tz_offset: i64) -> Result<&mut Self> {
+        if tz_offset <= -ONE_DAY || tz_offset >= ONE_DAY {
+            return Err(Error::unknown_timezone(tz_offset));
+        }
+        self.tz = match FixedOffset::east_opt(tz_offset as i32) {
+            None => return Err(Error::unknown_timezone(tz_offset)),
+            Some(tz) => tz,
+        };
+        Ok(self)
+    }
+
+    pub fn set_ignore_truncate(&mut self, new_value: bool) -> &mut Self {
+        self.ignore_truncate = new_value;
+        self
+    }
+
+    pub fn set_truncate_as_warning(&mut self, new_value: bool) -> &mut Self {
+        self.truncate_as_warning = new_value;
+        self
+    }
+
+    pub fn set_overflow_as_warning(&mut self, new_value: bool) -> &mut Self {
+        self.overflow_as_warning = new_value;
+        self
+    }
+
+    pub fn set_in_insert_stmt(&mut self, new_value: bool) -> &mut Self {
+        self.in_insert_stmt = new_value;
+        self
+    }
+
+    pub fn set_in_update_or_delete_stmt(&mut self, new_value: bool) -> &mut Self {
+        self.in_update_or_delete_stmt = new_value;
+        self
+    }
+
+    pub fn set_in_select_stmt(&mut self, new_value: bool) -> &mut Self {
+        self.in_select_stmt = new_value;
+        self
+    }
+
+    pub fn set_pad_char_to_full_length(&mut self, new_value: bool) -> &mut Self {
+        self.pad_char_to_full_length = new_value;
+        self
+    }
+
+    pub fn set_divided_by_zero_as_warning(&mut self, new_value: bool) -> &mut Self {
```

Comment:
Seems lots of the above functions are used only in test, should we add some comments or remove them and use flags instead?

Reply:
Readability